lumen/ai/agents.py

```diff
@@ -372,20 +372,17 @@ def _create_table_model(tables):
         return table_model
 
     async def answer(self, messages: list | str):
-        if len(memory["available_sources"]) >= 1:
-            available_sources = memory["available_sources"]
-            tables_to_source = {}
-            tables_schema_str = "\nHere are the table schemas\n"
-            for source in available_sources:
-                for table in source.get_tables():
-                    tables_to_source[table] = source
+        available_sources = memory["available_sources"]
+        tables_to_source = {}
+        tables_schema_str = "\nHere are the tables\n"
+        for source in available_sources:
+            for table in source.get_tables():
+                tables_to_source[table] = source
+                if isinstance(source, DuckDBSource) and source.ephemeral:
                     schema = get_schema(source, table, include_min_max=False, include_enum=True, limit=1)
```

Comment:
Getting schemas for all the tables in a IntakeDremioSource is pretty slow, so limit it to eph sources.